### PR TITLE
Feature/blog migratie

### PR DESCRIPTION
**Pull request by nicholasM95**
### Thank you for your contribution! Please go through this small checklist to prepare your pull request to be reviewed as soon as possible :)
* [ ] All authors were added to `_data/authors.yml` with a picture and an up-to-date bio
* [ ] The blog post image is of ratio 500x293
  * Feel free to leave a comment if you don't know how to resize your image and someone will gladly help you out
* [ ] The [**one sentence per line** method](https://raw.githubusercontent.com/brandon-rhodes/blog/master/texts/brandon/2012/one-sentence-per-line.rst) has been adopted
* [ ] A spell check has been done
* [ ] All links open in a new tab
  * Example: `[Amazing tech blog](https://ordina-jworks.github.io){:target="_blank" rel="noopener noreferrer"}`
* [ ] The post's content properly resizes according to the window's size
  * Tip: Just shorten your window's horizontal length to easily test this
  * To have images properly resize use `class="image fit"`
  * To have YouTube videos properly resize, wrap the iframe into a `<div class="responsive-video">`
* [ ] If a table of content is: check whether all links are working correctly

### Any specific remarks for the reviewers or anything you need help with?
\<enter text here>

### Checklist for the reviewers before merging
* [ ] Publishing date of the post has been correctly set
  * This can be done by renaming the filename or by setting the date in the header section of the post

### CODE REVIEW

File: _posts/2024-04-23-secure-blog-migration.md [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+> Discover how our migration from GitHub Pages to AWS Static Hosting not only bolsters our security with vital security headers, 
+> but also enhances user experience and SEO performance.
+
+- [About the JWorks tech blog](#about-the-jworks-tech-blog)
+- [Our previous situation](#our-previous-situation)
+- [The problem](#the-problem)
+- [Our solution](#our-solution)
+- [Action plan details](#action-plan-details)
+- [Result](#result)
+- [Conclusion](#conclusion)
```
Comment (denmette) [inline _posts/2024-04-23-secure-blog-migration.md]:
```suggestion
{:.no_toc}
- TOC
{:toc}
```

File: _posts/2024-04-23-secure-blog-migration.md [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+domain name and redirect users to our new website.
+
+## Action plan details
+We've used Terraform as Infrastructure as a Code combined with GitHub Actions to deploy all our resources.
+1. AWS WAF
+* Implemented **Rate limiting** to protect against DDoS Attacks.
+* Integrated Managed AWS Rules:
+  * **Common Rules** to protect against exploitation of a wide range of vulnerabilities including those described in OWASP publications.
+  * **Known Bad Input Rules** to block requests patterns that are known to be invalid and are associated with  exploitation or discovery of vulnerabilities.
+  * **Amazon Ip reputation List**  to block sources associated with bots or other threats.
```
Comment (denmette) [inline _posts/2024-04-23-secure-blog-migration.md]:
```suggestion
  * **Known Bad Input Rules** to block requests patterns that are known to be invalid and are associated with exploitation or discovery of vulnerabilities.
  * **Amazon IP reputation List**  to block sources associated with bots or other threats.
```

**Review by denmette — CHANGES_REQUESTED** [review]

File: _posts/2024-04-23-secure-blog-migration.md [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+We concluded that we faced a major problem with our security headers. We had none...
+After some research, we found that GitHub Pages doesn't provide an option to add these
+crucial security measures.
+
+<img src="{{ '/img/2024-04-23-secure-blog-migration/security_report_bad.png' | prepend: site.baseurl }}" alt="Bad Security Report" class="image fit" style="margin:0px auto; max-width:100%">
+
+### What are security headers
+Security headers are like special instructions that a website sends to your web 
+browser when you visit it. These instructions tell your browser how to behave to 
+keep you safe while you're on that website. They help protect against common online
```
Comment (pietervincken) [inline _posts/2024-04-23-secure-blog-migration.md]:
```suggestion
keep you safe while you're on a website. They help protect against common online 
```

File: _posts/2024-04-23-secure-blog-migration.md [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+After some research, we found that GitHub Pages doesn't provide an option to add these
+crucial security measures.
+
+<img src="{{ '/img/2024-04-23-secure-blog-migration/security_report_bad.png' | prepend: site.baseurl }}" alt="Bad Security Report" class="image fit" style="margin:0px auto; max-width:100%">
+
+### What are security headers
+Security headers are like special instructions that a website sends to your web 
+browser when you visit it. These instructions tell your browser how to behave to 
+keep you safe while you're on that website. They help protect against common online 
+threats like hackers trying to steal your information or take control of your browser.
```
Comment (pietervincken) [inline _posts/2024-04-23-secure-blog-migration.md]:
```suggestion
threats like hackers trying to steal your information or taking control of your browser.
```

File: _posts/2024-04-23-secure-blog-migration.md [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+
+## Our solution
+To address the security concerns, we made the decision to move our web hosting from GitHub
+Pages to a platform that allows us to configure security headers. This led us to migrate
+our blog to an Amazon S3 bucket within our AWS environment. Here we utilize AWS CloudFront
+to set up and enforce the necessary security headers.
+
+In addition, we've implemented an AWS WAF (Web Application Firewall) into our infrastructure
+to further enhance our website's security posture. Given our existing experience with AWS 
+through previous project, we decided to leverage the platform for its robust security features.
```
Comment (pietervincken) [inline _posts/2024-04-23-secure-blog-migration.md]:
```suggestion
through previous projects, we decided to leverage the platform for its robust security features.
```

File: _posts/2024-04-23-secure-blog-migration.md [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+
+By resolving this security issue, we seized the opportunity to migrate our blog 
+repository from GitHub Organization to GitHub Enterprise. 
+However, the old repository remains in place to preserve the original 
+domain name and redirect users to our new website.
+
+## Action plan details
+We've used Terraform as Infrastructure as a Code combined with GitHub Actions to deploy all our resources.
```
Comment (pietervincken) [inline _posts/2024-04-23-secure-blog-migration.md]:
```suggestion
We used Terraform as Infrastructure as a Code combined with GitHub Actions to deploy all our resources.
```

File: _posts/2024-04-23-secure-blog-migration.md (hunk 2 above)
Comment (pietervincken) [reply _posts/2024-04-23-secure-blog-migration.md]:
```suggestion
  * **Known Bad Input Rules** to block request patterns that are known to be invalid and are associated with  exploitation or discovery of vulnerabilities.
  * **Amazon Ip reputation List**  to block sources associated with bots or other threats.
```

File: _posts/2024-04-23-secure-blog-migration.md [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+* Integrated Managed AWS Rules:
+  * **Common Rules** to protect against exploitation of a wide range of vulnerabilities including those described in OWASP publications.
+  * **Known Bad Input Rules** to block requests patterns that are known to be invalid and are associated with  exploitation or discovery of vulnerabilities.
+  * **Amazon Ip reputation List**  to block sources associated with bots or other threats.
+
+2. AWS S3 Bucket
+* We created an S3 bucket that is fully private and only accessible by Cloudfront. 
+
+3. AWS ACM & Route 53
+* Created a subdomain for our blog using Route 53.
```
Comment (pietervincken) [inline _posts/2024-04-23-secure-blog-migration.md]:
```suggestion
* We created a subdomain for our blog using Route 53. 
```

File: _posts/2024-04-23-secure-blog-migration.md [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+  * **Common Rules** to protect against exploitation of a wide range of vulnerabilities including those described in OWASP publications.
+  * **Known Bad Input Rules** to block requests patterns that are known to be invalid and are associated with  exploitation or discovery of vulnerabilities.
+  * **Amazon Ip reputation List**  to block sources associated with bots or other threats.
+
+2. AWS S3 Bucket
+* We created an S3 bucket that is fully private and only accessible by Cloudfront. 
+
+3. AWS ACM & Route 53
+* Created a subdomain for our blog using Route 53. 
+* Generated a certificate via AWS ACM to ensure secure connections for our visitors.
```
Comment (pietervincken) [inline _posts/2024-04-23-secure-blog-migration.md]:
```suggestion
* We generated a certificate via AWS ACM to ensure secure connections for our visitors. 
```

File: _posts/2024-04-23-secure-blog-migration.md [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+2. AWS S3 Bucket
+* We created an S3 bucket that is fully private and only accessible by Cloudfront. 
+
+3. AWS ACM & Route 53
+* Created a subdomain for our blog using Route 53. 
+* Generated a certificate via AWS ACM to ensure secure connections for our visitors. 
+This certificate will be integrated into our CloudFront configuration.
+
+4. AWS Cloudfront
+* Initiated configuration of the origin, directing to our S3 bucket.
```
Comment (pietervincken) [inline _posts/2024-04-23-secure-blog-migration.md]: What is meant here? This is unclear for me.

File: _posts/2024-04-23-secure-blog-migration.md [hunk 10]
```diff
@@ -0,0 +1,151 @@
...
+* Following the setup of CloudFront security, we proceeded to connect our WAF
+* Implemented crucial security header, including:
+  * **Strict-Transport-Security (STS)** to enforce https
+  * **Content-Security-Policy (CSP)** to protect your site from XSS attacks. By whitelisting resources of approved content, you can prevent the browser from loading malicious assets
```
Comment (pietervincken) [inline _posts/2024-04-23-secure-blog-migration.md]:
```suggestion
  * **Content-Security-Policy (CSP)** to protect your site from XSS attacks. By whitelisting resources of approved content you can prevent the browser from loading malicious assets.
```

File: _posts/2024-04-23-secure-blog-migration.md [hunk 11]
```diff
@@ -0,0 +1,151 @@
...
+<img src="{{ '/img/2024-04-23-secure-blog-migration/security_report_good.png' | prepend: site.baseurl }}" alt="Good Security Report" class="image fit" style="margin:0px auto; max-width:100%">
+
+
+## Conclusion
+Our migration from GitHub Pages to AWS Static Hosting marks a significant step forward in bolstering our website's security, 
+enhancing user experience, and improving SEO performance. 
+By implementing essential security measures such as security headers and leveraging the robust infrastructure provided by AWS.
+We have fortified our defenses against potential threats, ensuring a safer online environment for both our users and our business. 
+Additionally, the transition enables us to deliver a smoother and more reliable user experience, enhancing trust and satisfaction among our audience.
+Moreover, optimization for SEO ensures improved visibility and accessibility, strengthening.
```
Comment (pietervincken) [inline _posts/2024-04-23-secure-blog-migration.md]:
```suggestion
Moreover, optimization for SEO ensures improved visibility and accessibility. 
```

File: _posts/2024-04-23-secure-blog-migration.md [hunk 12]
```diff
@@ -0,0 +1,151 @@
...
+
+## Result
+Through this transition to AWS infrastructure and the implementation of 
+comprehensive security measures, we've significantly enhanced the security 
+and reliability of our blog.
+
+<img src="{{ '/img/2024-04-23-secure-blog-migration/security_report_good.png' | prepend: site.baseurl }}" alt="Good Security Report" class="image fit" style="margin:0px auto; max-width:100%">
+
+
+## Conclusion
```
Comment (pietervincken) [inline _posts/2024-04-23-secure-blog-migration.md:136]: AI generated? It repeats the same point multiple time. Rewrite to make it more fluent and less repetitive.

File: _posts/2024-04-23-secure-blog-migration.md [hunk 13]
```diff
@@ -0,0 +1,151 @@
...
+Our migration from GitHub Pages to AWS Static Hosting marks a significant step forward in bolstering our website's security, 
+enhancing user experience, and improving SEO performance. 
+By implementing essential security measures such as security headers and leveraging the robust infrastructure provided by AWS.
+We have fortified our defenses against potential threats, ensuring a safer online environment for both our users and our business. 
+Additionally, the transition enables us to deliver a smoother and more reliable user experience, enhancing trust and satisfaction among our audience.
+Moreover, optimization for SEO ensures improved visibility and accessibility, strengthening. 
+With these enhancements in place, we are committed to providing a secure, seamless, and rewarding experience for all our users, 
+while maintaining the highest standards of security and performance.
+
+Curious if your website could use some enhancements too? Don't wait; take the first step and conduct a scan yourself on [Security Headers](https://securityheaders.com/){:target="_blank" rel="noopener noreferrer"}.
```
Comment (pietervincken) [inline _posts/2024-04-23-secure-blog-migration.md:142]: Excellent call to action! 👍

**Review by pietervincken — CHANGES_REQUESTED** [review]
Nice post, mostly cosmetic remarks. 
The conclusion needs significant work. Make sure to stress why we/you did it: summarise intro, summarise the outcome and highlight 1 interesting thing you learned in the process.

File: _posts/2024-04-23-secure-blog-migration.md (hunk 9 above)
Comment (ferrevangenechten1) [reply _posts/2024-04-23-secure-blog-migration.md]: The origins settings of CloudFront. 
image

But if it's already unclear for you, Il'll change it

**Review by yolanv — APPROVED** [review]
Fantastische blogpost mannen. Chapeau. Heel goed werk 👍

**Review by pietervincken — APPROVED** [review]

**Pull request merged**